extensions/kafka-connect/src/test/java/com/hazelcast/jet/kafka/connect/KafkaConnectDatagenIntegrationTest.java

```diff
@@ -87,5 +93,98 @@ public void testReadFromDatagenConnector() throws Exception {
+
+        try {
+            job.join();
+            fail("Job should have completed with an AssertionCompletedException, but completed normally");
+        } catch (CompletionException e) {
+            String errorMsg = e.getCause().getMessage();
+            assertTrue("Job was expected to complete with AssertionCompletedException, but completed with: "
+                    + e.getCause(), errorMsg.contains(AssertionCompletedException.class.getName()));
+        }
+    }
+
+
+    @Test
+    public void testSnapshotting() throws Exception {
+        int localParallelism = 3;
+        Properties randomProperties = new Properties();
+        randomProperties.setProperty("name", "datagen-connector");
+        randomProperties.setProperty("connector.class", "io.confluent.kafka.connect.datagen.DatagenConnector");
+        randomProperties.setProperty("tasks.max", Integer.toString(localParallelism));
+        randomProperties.setProperty("max.interval", "1");
+        randomProperties.setProperty("kafka.topic", "users");
+        randomProperties.setProperty("quickstart", "users");
+
+        Pipeline pipeline = Pipeline.create();
+        StreamStage<String> streamStage = pipeline.readFrom(KafkaConnectSources.connect(randomProperties))
+                .withoutTimestamps()
+                .setLocalParallelism(localParallelism)
+                .map(record -> Values.convertToString(record.valueSchema(), record.value()) + ", task.id: "
+                        + record.headers().lastWithName("task.id").value());
+        streamStage.writeTo(Sinks.logger());
+        streamStage
+                .writeTo(AssertionSinks.assertCollectedEventually(60,
+                        list -> assertEquals(localParallelism * ITEM_COUNT, list.size())));
+
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.addJarsInZip(new URL(CONNECTOR_URL));
+        jobConfig.setProcessingGuarantee(AT_LEAST_ONCE);
+        jobConfig.setSnapshotIntervalMillis(10);
+
+        Config config = smallInstanceConfig();
+        config.addMapConfig(new MapConfig("*")
+                .setEventJournalConfig(new EventJournalConfig().setEnabled(true))
+                .setBackupCount(3)
+        );
+        config.getJetConfig().setResourceUploadEnabled(true);
+        HazelcastInstance[] hazelcastInstances = createHazelcastInstances(config, 3);
+        JobRepository jobRepository = new JobRepository(hazelcastInstances[0]);
+        Job job = hazelcastInstances[0].getJet().newJob(pipeline, jobConfig);
+        assertJobStatusEventually(job, RUNNING);
+
+        waitForFirstSnapshot(jobRepository, job.getId(), 30, false);
+
+        hazelcastInstances[1].shutdown();
+
+        assertTrueEventually(() -> assertEquals(2, hazelcastInstances[0].getCluster().getMembers().size()));
+
+        try {
+            job.join();
+            fail("Job should have completed with an AssertionCompletedException, but completed normally");
```

Comment:
what if processing of incoming data end before job is restored? we do not check if job is still running after shutdown of the member

Reply:
added assertion https://github.com/hazelcast/hazelcast/pull/23831/commits/2d5e907883c5692205d7c168a69e68b839cfd728